### ISSUE TITLE
Use protobuf fields as native structs

### ISSUE THREAD

**Issue by L1ghtman2k** [ScoreTrak/ScoreTrak#230]
Currently, there is a translation layer between protobufs, and gorm's native structs: https://github.com/golang/protobuf/issues/52

It would be useful to instead consume protobuf models directly since it avoids unnecessary translations between the structs

Potential Approaches: 

- https://github.com/srikrsna/protoc-gen-gotag
- https://github.com/favadi/protoc-go-inject-tag
- wait for native protobuf support from golang
- tag module with https://github.com/gogo/protobuf

**Comment by L1ghtman2k** [comment]
https://github.com/alta/protopatch

**Comment by L1ghtman2k** [comment]
Examples with: https://github.com/favadi/protoc-go-inject-tag
https://github.com/liov/hoper
https://github.com/liyuliang/models

**Comment by L1ghtman2k** [comment]
Using just protoc-go-inect-tag seems to be harder than first anticipated. The custom protobuf field like: google.protobuf.BoolValue

It is possible to convert `.BoolValue` to optional bool instead, but there are still few types like Timespampts that are hard to convert. For now, I think this should wait a better alternative

**Comment by thisisibrahimd** [comment]
This appears to be a good options, https://github.com/infobloxopen/protoc-gen-gorm.

**Comment by thisisibrahimd** [comment]
The repo is actually under dev and not ready for prod

**Comment by L1ghtman2k** [comment]
> The repo is actually under dev and not ready for prod
very nice catch. I think at some point I have noticed that repo, but at the time it was only supporting gorm v1 (well, gorm v2 was very fresh back then). I think we let it incubate for a bit, and see how stable it gets.

**Comment by thisisibrahimd** [comment]
I don't believe there is a version 2 for gorm. https://pkg.go.dev/gorm.io/gorm?tab=versions

**Comment by L1ghtman2k** [comment]
So, i did a bit of digging, seems like gorm v2 was designated a different tag, from https://gorm.io/docs/v2_release_note.html:
// **NOTE** GORM `v2.0.0` released with git tag `v1.20.0`

So, the infobloxopen's repo uses `gorm v1.9.16`, which is still a v1 version from what i understand
